src/middlewared/middlewared/plugins/ups.py

```diff
@@ -191,10 +191,31 @@ def driver_choices(self):
         old_config['toemail'] = ';'.join(old_config['toemail']) if old_config['toemail'] else ''
 
         if len(set(old_config.items()) ^ set(config.items())) > 0:
+            if config['identifier'] != old_config['identifier']:
+                await self.dismiss_alerts()
+
             await self._update_service(old_config, config)
 
         return await self.config()
 
+    @private
+    async def alerts_mapping(self):
+        return {
+            'LOWBATT': 'UPSBatteryLow',
+            'COMMBAD': 'UPSCommbad',
+            'COMMOK': 'UPSCommok',
+            'ONBATT': 'UPSOnBattery',
+            'ONLINE': 'UPSOnline',
+            'REPLBATT': 'UPSReplbatt'
+        }
+
+    @private
+    async def dismiss_alerts(self):
+        config = await self.config()
+
+        for alert in (await self.alerts_mapping).values():
+            await self.middleware.call('alert.oneshot_delete', alert, {'ups': config['identifier']})
```

Comment:
We need to dismiss them for all UPSes.
Probably it's not possible with current alerts implementation and you might need to extend it a little.